Improve regex to eliminate ck/ik parameters in WWW-Authenticate

ben2610 opened the pull request:
The previous regex were not specific enough : I had the case where the nonce ended with ck, causing the regex to truncate the nonce and corrupt the format. This new regex will detect parameter boundary without ambiguity

ben2610 committed:
Improve regex to eliminate ck/ik parameters in WWW-Authenticate before forwarding register response to UE

ben2610 committed:
New HSS_REALM subst for icscf and scscf config

herlesupreeth commented:
@ben2610 thank you for the contribution.. currently i am on vacation will review once i am back

herlesupreeth commented on kamailio_icscf/icscf.cfg:
```diff
@@ -13,6 +13,7 @@ alias=ims.mnc001.mcc001.3gppnetwork.org
 #!define HOSTNAME "icscf.ims.mnc001.mcc001.3gppnetwork.org"
 
 #!subst "/NETWORKNAME/ims.mnc001.mcc001.3gppnetwork.org/"
+#!subst "/HSS_REALM/epc.mnc001.mcc001.3gppnetwork.org/"
```
please remove the changes related to HSS_REALM. I would like to keep the changes which correspond to only to regex to eliminate ck/ik parameters in WWW-Authenticate

ben2610 replied:
The rational behind this change is that the original tutorial to create an IMS system was based on a separate HSS for the IMS part, but with the latest open5gs code, it is no longer necessary to use a separate HSS: the HSS code of the open5gs project is sufficient. Hence, since the open5gs HSS is on the epc side, its realm will be epc... rather than ims...
If you still want to keep only the regex change, I will remove the HSS_REALM commit.

herlesupreeth replied:
I am aware that the open5gs HSS supports Cx interface. However, this repo is used as part of a tutorial in open5gs involving usage of FHoSS for Cx interface. So it would be great if you could just retain regex change and remove other changes. Many thanks for the extra effort.

ben2610 replied:
ok, but what about this: keeping HSS_REALM but give it default value ims... so that your script are still compatible with FHoHSS but become easily adaptable to open5gs HSS?

herlesupreeth commented on kamailio_icscf/kamailio_icscf.cfg:
```diff
@@ -163,7 +163,7 @@ modparam("cdp","config_file","/etc/kamailio_icscf/icscf.xml")
 modparam("ims_icscf", "cxdx_forced_peer", CXDX_FORCED_PEER)
 #!endif
-modparam("ims_icscf","cxdx_dest_realm", NETWORKNAME)
+modparam("ims_icscf","cxdx_dest_realm", "HSS_REALM")
```
same as above

herlesupreeth commented on kamailio_scscf/kamailio_scscf.cfg:
```diff
@@ -282,7 +282,7 @@ modparam("ims_auth", "registration_default_algorithm", REG_AUTH_DEFAULT_ALG)
 modparam("ims_auth", "cxdx_forced_peer", CXDX_FORCED_PEER)
 #!endif
-modparam("ims_auth", "cxdx_dest_realm", NETWORKNAME)
+modparam("ims_auth", "cxdx_dest_realm", "HSS_REALM")
```
same as above

herlesupreeth commented on kamailio_scscf/kamailio_scscf.cfg:
```diff
@@ -307,7 +307,7 @@ modparam("ims_registrar_scscf", "support_wildcardPSI",1)
 modparam("ims_registrar_scscf", "scscf_name", URI)
 modparam("ims_registrar_scscf", "scscf_name", URI)
-modparam("ims_registrar_scscf", "cxdx_dest_realm", NETWORKNAME)
+modparam("ims_registrar_scscf", "cxdx_dest_realm", "HSS_REALM")
```
same as above

herlesupreeth commented on kamailio_scscf/scscf.cfg:
```diff
@@ -14,6 +14,7 @@ listen=tcp:10.4.128.21:6060
 #!define URI "sip:scscf.ims.mnc001.mcc001.3gppnetwork.org:6060"
 
 #!subst "/NETWORKNAME/ims.mnc001.mcc001.3gppnetwork.org/"
+#!subst "/HSS_REALM/epc.mnc001.mcc001.3gppnetwork.org/"
```
same as above

herlesupreeth replied:
same as above

ben2610 committed:
Keep HSS_REALM but give it ims value by default

This will keep compatibility with the original scripts but making them
easily adaptable to HSS reside on the epc

ben2610 commented on kamailio_icscf/icscf.cfg:
```diff
@@ -13,6 +13,7 @@ alias=ims.mnc001.mcc001.3gppnetwork.org
 #!define HOSTNAME "icscf.ims.mnc001.mcc001.3gppnetwork.org"
 
 #!subst "/NETWORKNAME/ims.mnc001.mcc001.3gppnetwork.org/"
+#!subst "/HSS_REALM/ims.mnc001.mcc001.3gppnetwork.org/"
```
Hope this works for you.

Thanks and Best Regards,

herlesupreeth commented:
@ben2610 thank you very much for your contribution
